x_transformers/x_transformers.py

```diff
@@ -1913,6 +1915,15 @@ def __init__(
         self.emb_dim = emb_dim
         self.num_tokens = num_tokens
 
+        self.pooling = pooling
+        self.use_pooling = exists(pooling)
+        self.use_cls = use_cls
+        assert not (self.use_cls and self.use_pooling), 'cannot use both pooling and cls token'
+        if self.use_cls:
+            max_seq_len+=1
```

Comment:
instead of changing the `max_seq_len`, want to concat the cls token after adding the absolute positional embedding?